### PR TITLE
ath79: add support for the Netgear WNDRMAC series

### PR DESCRIPTION
**Pull request by cybik**
## Hardware support - WNDRMAC series

The Netgear WNDRMAC series are hardware variants of specific Netgear routers.

* The WNDRMAC v1 is a hardware variant of the Netgear WNDR3700v2 
  * See [commit](https://github.com/cybik/openwrt/commit/2db75642a25e2bc439baeb94974d73fb43633152) for the full spec, flash and information write-up
* The WNDRMAC v2 is a hardware variant of the Netgear WNDR3800 
  * See [commit](https://github.com/cybik/openwrt/commit/0a438f32d97199eeae0fbfb6266ed54131604ad1) for the full spec, flash and information write-up

They are considered near-clones of their "siblings", and there have been reports of being able to run the matching sibling firmware on the right target (wndr3700v2 on wndrmacv1, and wndr3800 on wndrmacv2).

## Test Report
* Latest binaries and config in use: [rel-2749971dd7 on my fork](https://github.com/cybik/openwrt/releases/tag/rel-2749971dd7)

### Tested
* Kernel: 5.4.42
* Latest source version: [r13521-2749971dd7](https://github.com/cybik/openwrt/commit/2749971dd789daa5ca6f77044a4bf600188aa8d6)
* Previously tested source versions:
  * [r13420-5971ca78ec](https://github.com/cybik/openwrt/commit/5971ca78ece8450b6d89c0e6fb9f5c182f98e3a7)

#### Wireless
* WiFi 2.4ghz Scan: *works* [as of r13521]
* WiFi 5ghz Scan: *works* [as of r13521]
* WiFi 2.4ghz Client into LAN and 5gHz as Master (relay mode): *worked once?* [as of r13521]

#### Buttons
* WiFi button "switcher"
  * With WiFi active, hitting button once disables WiFi [as of r13521]
  * With WiFi disabled, hitting button once enables WiFi [as of r13521]

#### LEDs
* Power LED: *works* [as of r13521]
* USB LED
  * USB port unused: powered down // *works as expected* [as of r13521]
  * USB port in use: lit up // *works as expected* [as of r13521]
* WiFi LED: *works* [as of r13521]
* WAN LED: *works* [as of r13521]
* Network activity LED: *works* [as of r13521]

### CODE REVIEW

**Comment by adrianschmutzler** [comment]
Please don't close one PR and create a new one just for changes. This will multiply our work in maintaining it, and will remove all discussion history. It would have been enough to just have two commits in one PR.

**Comment by cybik** [comment]
My apologies. I misunderstood the previous splitting comment.

**Comment by adrianschmutzler** [comment]
Okay, so now after you chose this one, let's keep it to this one, please. And maybe adjust the PR title.

**Comment by cybik** [comment]
I'm very sorry for the overeagerness.

As far as testing is concerned, at least for the v2, I may be able to validate that it'd work (I own a v2 variant). The v1 seems to be fairly rare at this time.

**Comment by adrianschmutzler** [comment]
So, you are a quick learner, looks almost perfect now.

Some nitpicks:
- "Add" in commit titles should be lower-case
- I'd replace the hyphen before the revision in commit title by a space (WNDRMAC-v2 -> WNDRMAC v2)

Despite, I'd still like to have specifications and flashing instructions in the commit message, like e.g. here:
https://github.com/openwrt/openwrt/commit/e66becb490796d0787f8bd2448ee6f14198a904a
https://github.com/openwrt/openwrt/commit/385f4868bc58b04e465db2fbcfce848a75009a74

Maybe you can get them from earlier Netgear commits or from the Wiki entries.

If you can't manage at all, use something like done here: https://github.com/openwrt/openwrt/commit/676ca94c3ca5421c7176d67a211fa29b1bb2e219

**Comment by adrianschmutzler** [comment]
And again, note that the only relevant information about the devices is what is stored in the commit message. Only this will get into the repository after all, while the comments in the PR (like you initial comment) will be forgotten soon ...

**Comment by cybik** [comment]
For what it's worth, I'm getting a spare WNDRMACv2 and will be testing a v2 flash.

If I can test only the v2, would it result in me having to rollback just the v1 patch so that we can get a v2 flash image without waiting for older hardware to be tested?

**Comment by adrianschmutzler** [comment]
If you test the v2 successfully, I'll just pick the v2 commit from this PR and will rebase it myself.

But be sure to check whether the device is fully functional, including LEDs, ports, WiFi, buttons etc.

**Comment by cybik** [comment]
My first test, flashing https://github.com/cybik/openwrt/releases/tag/proto%2Fath79 on a spare WNDRMACv2 I ordered, was successful.

Since my image was "dated" I wasn't able to `opkg install luci`, so I built building a new image based on latest-everything and pulling in a bunch of packages to test that most things work with LUCI.

* LEDs look like they're working as expected
* Preloading worked, LuCI is fine
* As-is, for whatever reason, WiFi is honked.

I'll try to fix the wifi before we go further.

**Comment by adrianschmutzler** [comment]
Well, without any additional information, nobody will be able to help you much ...

**Comment by cybik** [comment]
All right:

* Fundamentally speaking, I believe everything works except WiFi. Ethernet's good, device boots up into the new system just fine, any pre-loaded extensions/modules work
* Even the ath module gets loaded properly
* I installed lspci and hit a -vv, I can see the atheros eth wireless devices as well!
* Compared to a working ar71xx-based flash though, I believe that there is some PHY "customization" missing, maybe firmware not being dumped? 

My current idea is that bootup somehow doesn't hit firmware extraction either early enough, or at all, and this messes with how ath initializes itself *specifically* on this new device (the `10-ath9k-eeprom` part).

**Comment by cybik** [comment]
Correction: with a clean workspace on top of latest... I'm seeing the wifi on v2!

Going to test further.

**Comment by cybik** [comment]
I took the liberty of reordering the commits so that the first one (which is for the device I can test) can be picked and applied without much fuss.

@adrianschmutzler it's all yours!

File: package/boot/uboot-envtools/files/ath79 [hunk 1]
```diff
@@ -52,7 +52,8 @@ glinet,gl-ar150)
 	ubootenv_add_uci_config "/dev/mtd1" "0x0" "0x8000" "0x10000"
 	;;
 netgear,wndr3700|\
-netgear,wndr3700-v2)
+netgear,wndr3700-v2|\
+netgear,wndrmac-v2)
```
Comment (txt-file) [inline package/boot/uboot-envtools/files/ath79:57]: I have a wndr3800 running ath79-generic (OpenWrt master) which is not listed here. So either this entry is not needed or the wndr3800 is missing.

Comment (cybik) [reply package/boot/uboot-envtools/files/ath79:57]: I'm compiling a new payload with this line (and the corresponding v1, for the new commits) removed. I'll be uploading the new commits as soon as I validate this.

Comment (adrianschmutzler) [reply package/boot/uboot-envtools/files/ath79:57]: The wndr3800 is missing. However, that should be fixed separately if desired is not relevant here.

Comment (bobafetthotmail) [reply package/boot/uboot-envtools/files/ath79:57]: adding entries to uboot-envtools in the past was only accepted if the device actually needed to access the uboot envs at all. (like to reset a boot counter for failed boots in dual-partition devices or to read MAC address from uboot envs)

Comment (cybik) [reply package/boot/uboot-envtools/files/ath79:57]: All right then, I'll leave this in. Consider this PR as ready from my side.

**Pull request closed without merging**